### PR TITLE
Add growth rate <-> doubling time functions

### LINKED ISSUE #8: Rule of 72
Now and again I find myself wanting to convert from doubling time to growth rate and back. I normally [Rule of 72](https://en.wikipedia.org/wiki/Rule_of_72) this, but that gets inaccurate starting around growth rates ≥15%.

The formula for calculating this properly is simple but kinda annoying to remember and write repeatedly:

```
import math

def growth_rate_to_doubling_time(growth_rate):
    return math.log(2) / math.log(1 + growth_rate)

def doubling_time_to_growth_rate(doubling_time):
    return math.exp(math.log(2) / doubling_time) - 1

print(growth_rate_to_doubling_time(0.1)) # returns 7.27 -- 72/10 = 7.2
print(growth_rate_to_doubling_time(0.04)) # returns 17.67 -- 72/5 = 18
print(doubling_time_to_growth_rate(10)) # returns 0.072 -- 72/10 = 7.2%
print(doubling_time_to_growth_rate(3)) # returns 0.260 -- 72/10 = 24.0%
```

More problematically, I can't plug squigglepy distributions into these functions, since `math.log`/`math.exp` (and `np.log`/`np.exp`) don't work on distributions.

Tbh I'm not sure if this is a good fit for squigglpy as it has little to do with probabilities etc. (It's a better fit if you see squigglepy more as a general BOTEC/modelling framework.) But figured I'd raise it and let you decide.

### PR DESCRIPTION
**Pull request by erwald**
This closes #8.

I wasn't sure how to best fit it into the existing file structure, so definitely feel free to request changes.

Also, I chose to represent growth rates as fractions where e.g. 0.05 == +5%, but maybe it's more natural to let 1.05 == +5%? I could see it either way, anyway it's easy enough to change.

### CODE REVIEW

File: squigglepy/utils.py [hunk 1]
```diff
@@ -703,6 +703,71 @@ def laplace(s, n=None, time_passed=None,
...
+    -------
+    float or np.array or ComplexDistribution
+        Returns the doubling time.
+
+    Examples
+    --------
+    >>> growth_rate_to_doubling_time(0.01)
+    69.66071689357483
+    """
+    if is_dist(growth_rate):
```
Comment (peterhurford) [inline squigglepy/utils.py:730]: Great use of `is_dist` :)

File: squigglepy/utils.py [hunk 2]
```diff
@@ -703,6 +703,71 @@ def laplace(s, n=None, time_passed=None,
...
+        return math.log(2) / (1.0 + growth_rate).log()
+    elif isinstance(growth_rate, np.ndarray):
```
Comment (peterhurford) [inline squigglepy/utils.py:732]: Unsure but you could consider use `_is_numpy` from utils if you want to capture numpy things other than numpy arrays

File: squigglepy/utils.py [hunk 3]
```diff
@@ -703,6 +703,71 @@ def laplace(s, n=None, time_passed=None,
...
+    float or np.array or ComplexDistribution
+        Returns the growth rate expressed as a fraction (the percentage divided by 100).
+
+    Examples
+    --------
+    >>> doubling_time_to_growth_rate(12)
+    0.05946309435929531
+    """
+    if is_dist(doubling_time):
+        return (math.log(2) / doubling_time).exp() - 1
```
Comment (peterhurford) [inline squigglepy/utils.py:764]: Similar to above would rather have a squigglepy-speicfic `exp` or `dist_exp` function

File: tests/test_utils.py [hunk 4]
```diff
@@ -512,6 +514,36 @@ def test_laplace_no_time_passed():
...
+    assert doubling_time_to_growth_rate(1) == 1.0
+
+
+def test_doubling_time_to_growth_rate_nparray():
+    result = doubling_time_to_growth_rate(np.array([12, 5.5, 1]))
+    assert np.array_equal(result, np.array([0.05946309435929531, 0.13431252219546264, 1.0]))
+
+
+def test_doubling_time_to_growth_rate_dist():
+    assert doubling_time_to_growth_rate(const(12)) @ 1 == 0.05946309435929531
```
Comment (peterhurford) [inline tests/test_utils.py:544]: Tests look great - thanks!

File: CHANGES.md [hunk 5]
```diff
@@ -1,5 +1,8 @@
 ## v0.25
 
+* Added `log` and `exp` operators on distributions.
+* Added `growth_rate_to_doubling_time` and `doubling_time_to_growth_rate` convenience functions. These take numbers, numpy arrays or distributions.
```
Comment (peterhurford) [inline CHANGES.md:4]: Good use of `CHANGES.md`!

**Review by peterhurford — CHANGES_REQUESTED** [review]

**Comment by erwald** [comment]
@peterhurford Thanks, I've pushed some changes, lmk how it looks!

**Review by peterhurford — APPROVED** [review]

**Comment by peterhurford** [comment]
Thanks!

**Pull request merged**